File: golang/array-string/length-of-last-word.go
package main

func lengthOfLastWord(s string) int {
	pointer := len(s)
	result := 0
	for pointer > 0 {
		pointer--
		if s[pointer] != ' ' {
			result++
		} else if result > 0 {
			return result
		}
	}
	return result
}

func main() {
	output := lengthOfLastWord("Hello World")
	println(output) // Output: 5

	output = lengthOfLastWord("   fly me   to   the moon  ")
	println(output) // Output: 4

	output = lengthOfLastWord("luffy is still joyboy")
	println(output) // Output: 6
}

File: python/leetcode/array-string/length-of-last-word.py
def last_word_length(s):
    pointer = len(s) - 1
    length = 0

    while pointer >= 0:
        if s[pointer] != ' ':
            length += 1
            pointer -= 1
        else:
            if length > 0:
                return length
            pointer -= 1

    return length


if __name__ == '__main__':
    s = "Hello World"
    print(last_word_length(s))

    s = "   fly me   to   the moon  "
    print(last_word_length(s))


